go/libraries/doltcore/schema/encoding/schema_marshaling.go

```diff
@@ -130,8 +130,15 @@ func (enc encodedTypeInfo) decodeTypeInfo() (typeinfo.TypeInfo, error) {
 	return typeinfo.FromTypeParams(id, enc.Params)
 }
 
+type encodedIndex struct {
+	Name string `noms:"name" json:"name"`
+	Tags []uint64 `noms:"tags" json:"tags"`
+	Pks []uint64 `noms:"pks" json:"pks"`
```

Comment:
Consider restructuring how this is stored. The index is inherently tied to the schema, so it's odd to list the PKs separately here. A couple different options:

* Just store the tags of the indexed columns, including any Pks in the index. On index load, add any missing PKs from the underlying schema as necessary.
* Store the entire index schema of all indexed columns and any additional Pk columns. Then store an indexLen param tells you how many columns are in the index. Then you know that any tags past that length are primary keys, not part of the indexed columns.

I think I like the first option better?